codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -604,8 +604,7 @@ public void onUnknownFrame(
                 // Ignore unknown frames on connection stream, for example: HTTP/2 GREASE testing
                 return;
             }
-            onHttp2Frame(ctx, new DefaultHttp2UnknownFrame(frameType, flags, payload)
-                    .stream(requireStream(streamId)).retain());
+            onHttp2Frame(ctx, newHttp2UnknownFrame(frameType, streamId, flags, payload));
```

Comment:
Shouldn't the `requireStream(...)` be called in any case ? I mean if there is no stream its clearly an issue that should lead to a protocol error.

Reply:
moved to https://github.com/netty/netty/pull/14720/files#diff-26e15b3f312e851cc7d97a168303c880b10337845de48555a55d333f1aeccecfR715-R718

Reply:
https://github.com/netty/netty/pull/14720/files#diff-26e15b3f312e851cc7d97a168303c880b10337845de48555a55d333f1aeccecfR715-R718

Reply:
I know that's my point. We should always validate this, no matter what frame you create at the end.